model.py

```diff
@@ -109,7 +109,7 @@ class GitBranch(BaseGitSource):
     branch: NonEmptyStr = Field(..., description="Branch to check out")
 
 
-GitSource = GitRev | GitTag | GitBranch
+GitSource = GitRev | GitTag | GitBranch | BaseGitSource
```

Comment:
All types derive from `BaseGitSource` so its not required here. Or can you also have a situation where you dont specify `rev`, `tag`, or `branch` at all?

Reply:
yep there can be, I assumed it meant latest head. At least there was in the tests that @wolfv wrote for the PR.

Reply:
👍